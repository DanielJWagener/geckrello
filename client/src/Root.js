// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
