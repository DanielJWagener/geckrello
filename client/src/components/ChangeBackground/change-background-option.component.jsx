import React from "react";
import { connect } from "react-redux";

import { updateBoard } from "../../redux/boards/boards.actions";
import colorThemes from "../../utilities/colorThemes";
import updateBackgroundColor from "../../utilities/updateBackgroundColor";

export const ChangeBackgroundOption = props => {
  const theme = colorThemes[props.color];

  const updateBoardBackgroundColor = () => {
    updateBackgroundColor(theme);

    props.updateBoard(props.board._id, { background: props.color });
  };

  return (
    <div
      className="colors__option"
      style={{
        backgroundColor: theme.base
      }}
      onClick={updateBoardBackgroundColor}
    ></div>
  );
};

const mapStateToProps = ({ board }) => {
  return { board };
};

export default connect(mapStateToProps, { updateBoard })(
  ChangeBackgroundOption
);
