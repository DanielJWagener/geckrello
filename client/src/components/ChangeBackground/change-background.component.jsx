import React from "react";

import colorThemes from "../../utilities/colorThemes";
import ChangeBackgroundOption from "../ChangeBackground/change-background-option.component";

import "./change-background.styles.scss";

const colors = Object.keys(colorThemes);

const colorPalette = () =>
  colors.map(color => <ChangeBackgroundOption key={color} color={color} />);

const ChangeBackground = () => {
  return <div className="colors">{colorPalette()}</div>;
};

export default ChangeBackground;
