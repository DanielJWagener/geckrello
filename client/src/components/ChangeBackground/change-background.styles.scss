.colors {
  display: flex;
  flex-wrap: wrap;

  &__option {
    border-radius: 10px;
    height: 7.5rem;
    width: 14rem;
    margin: 1.5rem;
    cursor: pointer;
  }

  @media only screen and (max-width: 31.25em) {
    flex-direction: column;
    align-items: center;
  }
}
