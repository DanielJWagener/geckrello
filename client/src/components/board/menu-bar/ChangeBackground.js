import React from "react";
import ChangeBackgroundInner from "./ChangeBackgroundInner.js";

const ChangeBackground = props => {
  if (props.hidden) {
    return (
      <div className="sidebar-menu__interface sidebar-menu__interface--hidden">
        <ChangeBackgroundInner
          toggleChangeBackground={props.toggleChangeBackground}
        />
      </div>
    );
  } else {
    return (
      <div className="sidebar-menu__interface sidebar-menu__interface--visible">
        <ChangeBackgroundInner
          toggleChangeBackground={props.toggleChangeBackground}
        />
      </div>
    );
  }
};

export default ChangeBackground;
