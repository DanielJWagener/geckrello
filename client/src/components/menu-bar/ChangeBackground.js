import React from "react";
import ChangeBackgroundInner from "./ChangeBackgroundInner.js";

const ChangeBackground = ({ hidden, toggleChangeBackground }) => (
  <div
    className={`sidebar-menu__interface sidebar-menu__interface--${
      hidden ? "hidden" : "visible"
    }`}
  >
    <ChangeBackgroundInner toggleChangeBackground={toggleChangeBackground} />
  </div>
);

export default ChangeBackground;
