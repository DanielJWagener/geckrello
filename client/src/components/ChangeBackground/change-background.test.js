import { shallow } from "enzyme";
import React from "react";
import ChangeBackground from "./change-background.component";
import { ChangeBackgroundOption as ChangeBackgroundOptionPure } from "./change-background-option.component";
import Root from "../../Root";

it("renders ChangeBackgroundOption component", () => {
  expect(
    shallow(<ChangeBackgroundOptionPure color="blue" />).debug()
  ).toMatchSnapshot();
});

it("renders ChangeBackground component", () => {
  expect(shallow(<ChangeBackground />).debug()).toMatchSnapshot();
});
