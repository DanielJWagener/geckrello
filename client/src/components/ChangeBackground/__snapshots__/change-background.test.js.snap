// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ChangeBackground component 1`] = `
"<div className=\\"colors\\">
  <Connect(ChangeBackgroundOption) color=\\"purple\\" />
  <Connect(ChangeBackgroundOption) color=\\"green\\" />
  <Connect(ChangeBackgroundOption) color=\\"blue\\" />
  <Connect(ChangeBackgroundOption) color=\\"red\\" />
  <Connect(ChangeBackgroundOption) color=\\"orange\\" />
</div>"
`;

exports[`renders ChangeBackgroundOption component 1`] = `"<div className=\\"colors__option\\" style={{...}} onClick={[Function: updateBoardBackgroundColor]} />"`;
